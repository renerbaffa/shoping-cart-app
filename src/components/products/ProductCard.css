.ProductCard-container {
  width: 180px;
  border: 1px solid lightgray;
  margin: 8px 0px;
  position: relative;
}

.ProductCard-image {
  width: calc(100% - 12px);
  margin: 6px;
}

.ProductCard-name {
  width: calc(100% - 12px);
  margin: 6px;
  font-weight: bold;
  font-size: 14px;
}

.ProductCard-description {
  width: calc(100% - 12px);
  margin: 6px;
  font-size: 13px;
  margin-bottom: 40px;
}

.ProductCard-button {
  padding: 1px 4px;
  border-radius: 50%;
  position: absolute;
  right: 6px;
  bottom: 6px;
  background-color: #FF6D3E;
  color: white;
  cursor: pointer;
}
