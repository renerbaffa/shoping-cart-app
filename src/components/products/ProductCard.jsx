import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import Button from '../shared/Button';
import Image from '../shared/Image';

import './ProductCard.css';

const ProductCard = ({ className, description, image, name, onAddProduct, ...other }) => (
  <div className={cx('ProductCard-container', className)} {...other}>
    <Image
      className="ProductCard-image"
      alt={name}
      src={image}
    />
    <div className="ProductCard-name">{name}</div>
    <div className="ProductCard-description">{description}</div>
    <Button
      className="ProductCard-button"
      onClick={onAddProduct}
    >
      +
    </Button>
  </div>
);

ProductCard.propTypes = {
  className: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  name: PropTypes.string,
  onAddProduct: PropTypes.func,
};

ProductCard.defaultProps = {
  className: '',
  description: '',
  image: '',
  name: '',
  onAddProduct: () => {},
};

export default ProductCard;
