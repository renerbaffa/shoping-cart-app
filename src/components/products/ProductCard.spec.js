import React from 'react';
import renderer from 'react-test-renderer';

import ProductCard from './ProductCard';

describe('<ProductCard />', () => {
  it('should render correctly', () => {
    const productCard =
      renderer
        .create(
          <ProductCard
            className="className"
            description="Product name sample"
            image="http://lorempixel.com/400/200/technics/"
            name="Product name sample"
          />
        )
        .toJSON();
    expect(productCard).toMatchSnapshot();
  });

  it('should reach 100%', () => {
    ProductCard.defaultProps.onAddProduct();
  });
});
