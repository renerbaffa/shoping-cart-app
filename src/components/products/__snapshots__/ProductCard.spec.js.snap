// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render correctly 1`] = `
<div
  className="ProductCard-container className"
>
  <img
    alt="Product name sample"
    className="ProductCard-image"
    src="http://lorempixel.com/400/200/technics/"
  />
  <div
    className="ProductCard-name"
  >
    Product name sample
  </div>
  <div
    className="ProductCard-description"
  >
    Product name sample
  </div>
  <button
    className="ProductCard-button Button"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
